docs/auth/add-auth-provider.md

```diff
@@ -122,191 +121,128 @@ due to its comprehensive set of supported authentication
 
-### Installing the dependencies:
+Create a new module using `yarn new`, pick `backend-module` and provide `auth-backend` as the plugin ID and `providername-provider` as the module ID.
+
+Make sure that the module has the appropriate passport provider as a dependency.
 
 ```bash
-cd plugins/auth-backend
+cd plugins/auth-backend-backend-module-providername-provider
 yarn add passport-provider-a
 yarn add @types/passport-provider-a
 ```
 
-### Create implementation
-
-Make a new folder with the name of the provider following the below file
-structure:
-
-```bash
-plugins/auth-backend/src/providers/providerA
-├── index.ts
-└── provider.ts
-```
-
-**`plugins/auth-backend/src/providers/providerA/provider.ts`** defines the
-provider class which implements a handler for the chosen framework.
-
 ### Adding an OAuth based provider
 
-If we're adding an `OAuth` based provider we would implement the
-`OAuthHandlers` interface. By implementing this
-interface we can use the `OAuthProvider` class provided by `lib/oauth`, meaning
-we don't need to implement the full
-`AuthProviderRouteHandlers` interface that providers
-otherwise need to implement.
-
-The provider class takes the provider's options as a class parameter. It also
-imports the `Strategy` from the passport package.
-
-```ts
-import { Strategy as ProviderAStrategy } from 'passport-provider-a';
-
-export type ProviderAProviderOptions = OAuthProviderOptions & {
-  // extra options here
-}
-
-export class ProviderAAuthProvider implements OAuthHandlers {
-  private readonly _strategy: ProviderAStrategy;
-
-  constructor(options: ProviderAProviderOptions) {
-    this._strategy = new ProviderAStrategy(
-      {
-        clientID: options.clientId,
-        clientSecret: options.clientSecret,
-        callbackURL: options.callbackUrl,
-        passReqToCallback: false,
-        response_type: 'code',
-        /// ... etc
-      }
-      verifyFunction, // See the "Verify Callback" section
-    );
-  }
-
-  async start() {}
-  async handler() {}
-}
-```
-
-### Adding an non-OAuth based provider
-
-An non-`OAuth` based provider could implement
-`AuthProviderRouteHandlers` instead.
-
-```ts
-type ProviderAOptions = {
-  // ...
-};
-
-export class ProviderAAuthProvider implements AuthProviderRouteHandlers {
-  private readonly _strategy: ProviderAStrategy;
-
-  constructor(options: ProviderAOptions) {
-    this._strategy = new ProviderAStrategy(
-      {
-        // ...
+We're then creating a new module that can extend the Auth backend using the `authProvidersExtensionPoint`.
+
+```ts title="plugins/auth-backend-providername-provider/src/module.ts"
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import {
+  authProvidersExtensionPoint,
+  commonSignInResolvers,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import { providerAuthenticator } from './authenticator';
+
+/** @public */
+export const authModuleOktaProvider = createBackendModule({
```

Comment:
Thinking it might be helpful to pick an example provider name, for example "my-example", and use that throughout the examples. It's probably quite easy to confuse some of the naming patterns here otherwise. So you'd have for example `plugins/auth-backend-my-example-provider/src`, `authModuleMyExampleProvider` here, `myExampleAuthenticator`, etc.

Reply:
Was going for providername but i agree that it's too generic. Also that cost need a new name. Will fix.